### ISSUE TITLE
Type mismatching with @types/markdown-it 14.1.2

### ISSUE THREAD

**Issue by Fachep** [vuejs/vitepress#4116]
### Describe the bug

DefinitelyTyped/DefinitelyTyped#70083 changed type `Delimiter`.
Now npm resolves `"@types/markdown-it": "^14.1.1"` to 14.1.2, which does not match the type definition exported from vitepress.
Adding plugins for MarkdownIt causes tsc error TS2769: No overload matches this call.

### Reproduction

package.json:
```json
{
  "type": "module",
  "devDependencies": {
    "vitepress": "1.3.2",
    "@types/markdown-it": "^14.1.1",
    "typescript": "latest"
  }
}
```
https://stackblitz.com/edit/vite-qlr8wt?file=tsc.out.log
Run `npx tsc` at package root.

### Expected behavior

No errors from tsc.

### System Info

```Text
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.20.3 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    @types/markdown-it: ^14.1.1 => 14.1.2 
    vitepress: 1.3.2 => 1.3.2
```
```


### Additional context

Specifying version 14.1.1 for `@types/markdown-it` in package.json works for now.
```typescript
// node_modules/vitepress/dist/node/index.d.ts

interface Delimiter {
    marker: number;
    length: number;
    jump: number;
    token: number;
    end: number;
    open: boolean;
    close: boolean;
}
```
I guess vitepress shouldn't embed external type definitions like this?

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment]
Add ts-ignore for now. It will be fixed when we update deps here.

**Comment by Trinovantes** [comment]
Is there any reason why VitePress is currently bundling `@types/markdown-it` in `dist/node/index.d.ts`?

```
$ grep "interface MarkdownIt" node_modules/vitepress/dist/node/index.d.ts
1250:interface MarkdownItConstructor {
1259:interface MarkdownIt {
1712:    interface MarkdownItEnv {
1731:interface MarkdownItHeader {
1794:    interface MarkdownItEnv {
1868:    interface MarkdownItEnv {
```

This problem will keep happening on every future `@types/markdown-it` version. It would be nice to just extern this dependency so even if VitePress and plugins use different versions, we can use our package manager to override the resolved version.

**Comment by peterroe** [comment]
Now its black coal kiln has been restored, right?

**Comment by brc-dd** [comment]
just released in 1.3.3
